src/modules/shared/utils.ts

```diff
@@ -0,0 +1,40 @@
+import { COUNTRY_LOCALES } from './data/countryLocales';
+
+export const getLocale = (): string => {
+  // Define a list of supported locales.
+  const supportedLocales = Object.values(COUNTRY_LOCALES);
+
+  // Check if running in a non-browser environment (e.g., Node.js or older browsers).
+  if (typeof navigator === 'undefined') {
+    return 'en-US';
+  }
+
+  // Check if the browser supports the Intl object and user language preferences.
+  if (window.Intl && typeof window.Intl === 'object') {
+    const userLocales = window.navigator.languages || [
+      window.navigator.language,
+    ];
+
+    for (const locale of userLocales) {
+      const normalizedLocale = supportedLocales.find((supportedLocale) =>
+        locale.toLowerCase().startsWith(supportedLocale.toLowerCase()),
```

Comment:
Why do we need supportedLocales ? 
If `window.navigator.language` is absent, can we simply not take `window.navigator.languages[0]` as this array is sorted in order of preference.
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/languages

Adding supportedLocales will only increase our bundle size for every numberformat utility.